modules/weaver-tests/src/smithy4s/weavertests/WeaverTests.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s.weavertests
+
+import cats.effect.IO
+import org.http4s.HttpApp
+import org.http4s.HttpRoutes
+import org.http4s.Uri
+import smithy.test._
+import smithy4s.Document
+import smithy4s.Endpoint
+import smithy4s.Service
+import smithy4s.ShapeTag
+import smithy4s.UnsupportedProtocolError
+import weaver._
+
+case class GeneratedTest(name: String, assertions: IO[Expectations])
+
+abstract class WeaverTests[P, Alg[_[_, _, _, _, _]], Op[_, _, _, _, _]](
+    protocolTag: ShapeTag[_],
+    client: (
+        HttpApp[IO],
+        Uri
+    ) => Either[UnsupportedProtocolError, smithy4s.Monadic[Alg, IO]],
+    server: smithy4s.Monadic[Alg, IO] => Either[
```

Comment:
the client and server classes should be separate classes : protocol implementors might not be able to implement both sides, and they still need to be able to benefit from it